packages/ui/src/Forms/CustomerProfile/CustomerProfileForm.tsx

```diff
@@ -0,0 +1,319 @@
+                            e.target.value.replace(/\s/g, "")
+                          )
+                        }
+                        disabled={!isEditing}
+                      />
+                    )}
+                  </Field>
+                </div>
+              </div>
+            </Section>
+
+            <Section
+              title="Medical Details"
+              subtitle="Manage your medical details"
+            >
+              <div className="grid sm:grid-cols-6 gap-y-2">
+                <div className="col-span-4">
+                  <Field name="certificateExpiration">
+                    {(field: FieldProps) => (
+                      <DateInput
+                        formikField={field}
+                        label={t(CustomerLabel.CertificateExpiration)}
+                        StartAdornment={
+                          <IconAdornment
+                            Icon={<ClipboardList />}
+                            position="start"
+                            disabled={!isEditing}
+                          />
+                        }
+                        disabled={!isEditing}
+                      />
+                    )}
+                  </Field>
+                </div>
+                <div className="col-span-4">
+                  <Field name="covidCertificateReleaseDate">
+                    {(field: FieldProps) => (
+                      <DateInput
+                        formikField={field}
+                        label={t(CustomerLabel.CovidCertificateReleaseDate)}
+                        StartAdornment={
+                          <IconAdornment
+                            Icon={<ShieldCheck />}
+                            position="start"
+                            disabled={!isEditing}
+                          />
+                        }
+                        disabled={!isEditing}
+                      />
+                    )}
+                  </Field>
+                </div>
+                <div className="col-span-4">
+                  <Field name="covidCertificateSuspended" type="checkbox">
+                    {(field: FieldProps) => (
+                      <Checkbox
+                        formikField={field}
+                        label={t(CustomerLabel.CovidCertificateSuspended)}
+                        helpText="Check this box if your COVID certificate is more than 9 months old"
+                        disabled={!isEditing}
+                      />
+                    )}
+                  </Field>
+                </div>
+              </div>
+            </Section>
+
+            {isEditing ? (
+              <>
+                <div className="flex justify-self-end gap-x-2 mt-5">
+                  <Button
+                    type="reset"
+                    onClick={() => {
+                      onCancel();
+                      resetForm();
+                      toggleEdit();
+                    }}
```

Comment:
All of this `onClick` functionality can be written up in `Formik`'s `onReset` and here, you just use `type="reset"` to trigger it (I think it would require `Form`, as mentioned above, though. 🤔

Reply:
I've spent a hot second already trying to get this to work and but for some reason it won't wire up the "handleReset" to the button with `type='reset'` when using Form. Going to let this lie until the next round of work